src/suite.py

```diff
@@ -533,36 +450,78 @@ def all_combinations(self, function, matched_pairs=False):
 
   # training the model
   def self_issue_classification_all(self, matched_pairs=False):
-
-    def self_issue_classification_statistics_per():
-      # self.train_data
-      self.self_issue_classification_from_comments()
-      score = fbeta_score(
-          self.train_data["prediction"].tolist(),
-          self.train_data["label"].tolist(),
-          average="weighted",
-          beta=0.5)
-
-      return score
-
-    best_score = self.all_combinations(
-        self_issue_classification_statistics_per, matched_pairs=matched_pairs)
-    logging.info("Best f0.5 score: {}".format(best_score))
-    return self.model
+    global model
+    # n-fold nested cross validation
+    # https://scikit-learn.org/stable/auto_examples/model_selection/plot_nested_cross_validation_iris.html
+    svm = SVC()
+    num_trials = 5
+    n_splits = 10
+    best_model = None
+    best_score = 0
+    best_ratio = 0
+    for ratio in self.ratio:
+      for i in range(num_trials):
+        # find the best paramter combination
+        # need to try different ratios later
+        train_data = util.select_ratio(self.all_train_data, ratio)
+        y_train = train_data["label"]
+        X_train = train_data[self.features]
+        model = GridSearchCV(
+            estimator=svm,
+            param_grid=self.param_grid,
+            scoring="f1_weighted",
+            n_jobs=14,  # parallel
+            cv=KFold(n_splits=n_splits, shuffle=True),
+            verbose=0)
+        model.fit(X_train, y_train)
+
+        # nested cross validation with paramter optimization
+        nested_score = model_selection.cross_val_score(
+            model, X_train, y_train, cv=KFold(n_splits=n_splits, shuffle=True))
+        nested_scores = nested_score.mean()
+        if nested_scores > best_score:
+          best_score = nested_scores
+          best_model = model
+          best_ratio = ratio
+
+    # Find the optimal parameters
+    logging.info("Trying all combinations of hyper parameters.")
+    logging.info("Scores with {}-fold cross validation".format(n_splits))
+    logging.info("Best parameter: {}, ratio: {}.".format(
+        best_model.best_estimator_, best_ratio))
+    logging.info("Best score: {}.".format(best_model.best_score_))
+    self.model = best_model
+
+    #self.cross_validation()
+
+    #logging.info("Removing angry words towards oneself and SE words.")
+    y_train = self.all_train_data["label"]
+    X_train = self.all_train_data[self.features]

```

Comment:
lowecase?

Reply:
also you've got a commetned out log statement

Reply:
> lowecase?

you mean X_train? I'd like to leave it as X_train, like in the tutorials I followed - if I needed to copy their code, it'd been easier.